File: Java/src/test/java/com/pduda/field/ShoppingCartTest.java
package com.pduda.field;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;


public class ShoppingCartTest {


    @Test
    public void singleItem_numberOfProductsInTheCart() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.add(10);

        assertThat( shoppingCart.numberOfProducts()).isEqualTo(1);
    }

    @Test
    public void singleItem_totalPrice() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.add(10);

        assertThat(shoppingCart.calculateTotalPrice()).isEqualTo(10);
    }

    @Test
    public void singleItem_hasDiscountIfContainsAtLeastOneProductWorthAtLeast100() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.add(100);

        assertThat(shoppingCart.hasDiscount()).isTrue();
    }

    @Test
    public void singleItem_doesNotHaveDiscountIfContainsNoProductsWorthAtLeast100() throws Exception {
        ShoppingCart shoppingCart = new ShoppingCart();
        shoppingCart.add(99);

        assertThat(shoppingCart.hasDiscount()).isFalse();
    }

}

File: Kotlin/src/test/kotlin/com/pduda/field/ShoppingCartTest.kt
package com.pduda.field

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

class ShoppingCartTest {
    @Test
    @Throws(Exception::class)
    fun singleItem_numberOfProductsInTheCart() {
        val shoppingCart = ShoppingCart()
        shoppingCart.add(10)
        Assertions.assertThat(shoppingCart.numberOfProducts()).isEqualTo(1)
    }

    @Test
    @Throws(Exception::class)
    fun singleItem_totalPrice() {
        val shoppingCart = ShoppingCart()
        shoppingCart.add(10)
        Assertions.assertThat(shoppingCart.calculateTotalPrice()).isEqualTo(10)
    }

    @Test
    @Throws(Exception::class)
    fun singleItem_hasDiscountIfContainsAtLeastOneProductWorthAtLeast100() {
        val shoppingCart = ShoppingCart()
        shoppingCart.add(100)
        Assertions.assertThat(shoppingCart.hasDiscount()).isTrue
    }

    @Test
    @Throws(Exception::class)
    fun singleItem_doesNotHaveDiscountIfContainsNoProductsWorthAtLeast100() {
        val shoppingCart = ShoppingCart()
        shoppingCart.add(99)
        Assertions.assertThat(shoppingCart.hasDiscount()).isFalse
    }
}